app/uk.gov.hmrc.helptosavestridefrontend/controllers/SessionBehaviour.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosavestridefrontend.controllers
+
+import java.util.UUID
+
+import cats.instances.future._
+import play.api.libs.json.{Format, Json}
+import play.api.mvc.{Request, Result, Session}
+import uk.gov.hmrc.helptosavestridefrontend.connectors.KeyStoreConnector
+import uk.gov.hmrc.helptosavestridefrontend.models.PayePersonalDetails
+import uk.gov.hmrc.helptosavestridefrontend.models.eligibility.EligibilityCheckResult
+import uk.gov.hmrc.helptosavestridefrontend.models.eligibility.EligibilityCheckResult.{AlreadyHasAccount, Eligible, Ineligible}
+import uk.gov.hmrc.helptosavestridefrontend.util.{Logging, toFuture}
+import uk.gov.hmrc.helptosavestridefrontend.views
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+trait SessionBehaviour {
+  this: StrideFrontendController with Logging ⇒
+
+  val keyStoreConnector: KeyStoreConnector
+
+  val sessionKey: String = "stride-user-info"
+
+  def checkSession(noSession: ⇒ Future[Result], whenSession: UserInfo ⇒ Future[Result])(implicit request: Request[_]): Future[Result] = {
+
```

Comment:
if there is no key in the session we can actually avoid making a call to keystore. If instead of `getSessionWithKey` we had something like:
```scala
def getSessionKey(session: Session): Option[String] = session.get(sessionKey)
```
then `checkSession` could look something like this:
```scala
def checkSession(noSession: ⇒ Future[Result], whenSession: UserInfo ⇒ Future[Result])(implicit request: Request[_]): Future[Result] =
  getSessionKey(request.session) match {
    case Some(key) =>
      keyStoreConnector.get(key).fold...
    case None =>
      noSession
  }
```

Reply:
we only want to generate a new key when the operator is at the start of the journey, if the key doesn't exist we want to redirect them to the start page